tests/functions/module_lookup.test.ts

```diff
@@ -1922,3 +1922,296 @@ describe("OFFSET formula", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=DROP(A1:B2,-2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("#ERROR");
+  });
+  test("error if too many columns to exclude in negative", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=DROP(A1:B2,,-2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("#ERROR");
+  });
+});
+
+describe("ARRAYTOTEXT formula", () => {
+  test("correct without format", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=ARRAYTOTEXT(A1:B2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("A1,B1,A2,B2");
+  });
+  test("correct with format 1", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=ARRAYTOTEXT(A1:B2,1)",
+    };
+    expect(evaluateCell("D5", grid)).toBe('{"A1","B1";"A2","B2"}');
+  });
+  test("error with wrong format", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=ARRAYTOTEXT(A1:B2,2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("#ERROR");
+  });
+  test("correct with an empty cell", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1",
+      A2: "A2", B2: "B2",
+                                    D5: "=ARRAYTOTEXT(A1:B2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("A1,,A2,B2");
+  });
+  test("correct with links, booleans, decimals and errors", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "www.odoo.com", B1: "true", C1: "12/12/2012",
+      A2: "3.14", B2: "#ERROR", C2: "",
+                                    D5: "=ARRAYTOTEXT(A1:C2)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("www.odoo.com,TRUE,41255,3.14,#ERROR,");
+  });
+});
+
+describe("FORMULATEXT formula", () => {
+  test("correct with a formula", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "=SUM(B1:B2)", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=FORMULATEXT(A1)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("=SUM(B1:B2)");
+  });
+  test("correct with a classic cell", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "A1", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=FORMULATEXT(A1)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("#N/A");
+  });
+  test("correct with an empty cell", () => {
+    //prettier-ignore
+    const grid = {
+      A1: "", B1: "B1",
+      A2: "A2", B2: "B2",
+                                    D5: "=FORMULATEXT(A1)",
+    };
+    expect(evaluateCell("D5", grid)).toBe("#N/A");
+  });
+});
+describe("TAKE formula", () => {
```

Comment:
You can also add a test for the edge case where everything should be taken: `=TAKE(A1:B2,,)